Fix bug where clicking first track does not play

zachmandeville opened the pull request:
Previously, upon initial page load, if you click the first track in a
playlist, the UI would indicate that the song is playing but no music
would play.

This was because state.trackIndex is initialized to 0, so when we go
into play(), datradio would assume that the first track was already
playing, and then call resumeTrack instead of playTrack.

Now, we initialize state.trackIndex to null, such that clicking the
first track will correctly begin playing rather than attempting to
resume.

Commit:
Fix bug where clicking first track does not play

Previously, upon initial page load, if you click the first track in a
playlist, the UI would indicate that the song is playing but no music
would play.

This was because state.trackIndex is initialized to 0, so when we go
into play(), datradio would assume that the first track was already
playing, and then call resumeTrack instead of playTrack.

Now, we initialize state.trackIndex to null, such that clicking the
first track will correctly begin playing rather than attempting to
resume.

cblgh commented:
:000000 the first PR! 🔥 🔥 🔥 

Good catch! I'm a bit wary of having null in place of a number, weird things can happen when you e.g. do `parseInt(null) // gives us NaN`

Think you can update the PR such that the initialization is changed from `null` to `-1` instead? I think that would have the same effect, except with maybe better behaviour when e.g. the next track hotkey is pressed!

web-netizen committed:
change initial state.track.index from null to -1

Based on feedback from cblgh, modified my initial pull request so that state.track.index initializes as -1 instead of as Null.  This change would also fix the issue where the first track does not play, while also ensuring other functions (like the hot key for next track) still work.

zachmandeville commented:
This makes sense!  I've updated the pull request!

cblgh commented:
yay!
